pkg/operator/staticpod/controllers.go

```diff
@@ -105,6 +106,10 @@ type Builder interface {
 	WithCustomInstaller(command []string, installerPodMutationFunc installer.InstallerPodMutationFunc) Builder
 	WithPruning(command []string, staticPodPrefix string) Builder
 	WithPodDisruptionBudgetGuard(operatorNamespace, operatorName, readyzPort, readyzEndpoint string, createConditionalFunc func() (bool, bool, error)) Builder
+	// WithAlwaysAllowUnhealthyPodEvictionPolicy allows eviction of unhealthy (not ready) pods even if there are no disruptions allowed on a PodDisruptionBudget.
+	// This can help to drain/maintain a node and recover without a manual intervention when multiple instances of nodes or pods are misbehaving.
+	// Use this with caution, as this option can disrupt perspective pods that have not yet had a chance to become healthy.
+	WithAlwaysAllowUnhealthyPodEvictionPolicy() Builder
```

Comment:
- This is meant to be an opt-in, but it can be also added to `WithPodDisruptionBudgetGuard` options.
- I am not allowing passing `UnhealthyPodEvictionPolicyType` reference, because we cannot be sure if new values are added later to the API in the future that could disrupt high availability. The guarding  of the pods should be tested before allowing it.

Reply:
@atiratree let's move it to `WithPodDisruptionBudgetGuard`, since that's the PDB you are modifying here, so that seems like the best API for this.

Reply:
done, and using a UnhealthyPodEvictionPolicyType type now. I am also testing this change again in https://github.com/openshift/cluster-kube-apiserver-operator/pull/1579